frontend/src/components/datasets/dataset-access-icon/dataset-access-icon.tsx

```diff
@@ -0,0 +1,39 @@
+import { EyeInvisibleOutlined } from '@ant-design/icons';
+import { Popover } from 'antd';
+import clsx from 'clsx';
+import { useTranslation } from 'react-i18next';
+
+import shieldHalfIcon from '@/assets/icons/shield-half-icon.svg?react';
+import { CustomSvgIconLoader } from '@/components/icons/custom-svg-icon-loader/custom-svg-icon-loader.component';
+import { DatasetAccess } from '@/types/dataset';
+import { getDatasetAccessTypeLabel } from '@/utils/access-type.helper';
+
+import styles from './dataset-access-icon.module.scss';
+
+type Props = {
+    accessType: DatasetAccess;
+    hasPopover?: boolean;
+};
+
+export const DatasetAccessIcon = ({ accessType, hasPopover = false }: Props) => {
+    const { t } = useTranslation();
```

Comment:
I would work with a switch statement here, that allows easier extension if we have more types in the future